packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -1123,7 +1123,14 @@ class RenderListWheelViewport
   }
 
   @override
-  RevealedOffset getOffsetToReveal(RenderObject target, double alignment, { Rect? rect }) {
+  RevealedOffset getOffsetToReveal(
+    RenderObject target,
+    double alignment, {
+    Rect? rect,
+    AxisDirection? axisDirection,
+  }) {
+    // ListWheelViewport only support vertical axes.
+    assert(axisDirection == null || axisDirectionToAxis(axisDirection) == Axis.vertical);
```

Comment:
After reading through this PR, I wonder if instead of adding the axis param (and then somewhat awkwardly asserting whether you passed in the right one (for 1D scrollers) or any axis at all (for 2D scrollers)), we could "just" extend the return type RevealedOffset to include a second offset. The existing `RevealedOffset.offset` would be the offset for the mainAxis (= only axis for 1D scrollers and whatever was declared as main for 2D scrollers) and the newly added `notMainAxisOffset` property (name TBD) would be either 0 for 1D scrollers or the offset required to reveal the thing in the other axis for 2D scrollers.

Reply:
It would still need to be called twice since is is called for the individual scroll position of each axis. If it returns two offsets, the vertical scroll position cannot do anything about the horizontal offset that would be returned if we returned both.